src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringMethodCharOverloadWithSingleCharacters.cs

```diff
@@ -0,0 +1,209 @@
+
+                // CA1865: Method(string, StringComparison.Ordinal) or
+                //         Method(ascii string, StringComparison.InvariantCulture)
+                if (comparison == StringComparisonUsed.Ordinal ||
+                    (comparison == StringComparisonUsed.InvariantCulture && c.IsAscii()))
+                {
+                    rule = SafeTransformationRule;
+                }
+                // CA1866: Method(string)
+                else if (comparison == null)
+                {
+                    rule = NoSpecifiedComparisonRule;
+                }
+                // CA1867: Method(string, StringComparison.AnythingElse)
+                else
+                {
+                    rule = AnyOtherSpecifiedComparisonRule;
+                }
+
+                var argumentList = GetArgumentList(stringArgument.Syntax);
+                if (argumentList != null)
+                {
+                    var arg0 = $"string.{method}(char)";
+                    var arg1 = $"string.{method}(string)";
+                    context.ReportDiagnostic(argumentList.CreateDiagnostic(rule, new[] { arg0, arg1 }));
+                }
+            }
+
+            bool TryMatchTargetMethod(
+                IInvocationOperation invocationOperation,
+                [NotNullWhen(true)] out string? method,
+                out StringComparisonUsed? comparison)
+            {
+                method = null;
+                comparison = null;
+
+                if (invocationOperation.TargetMethod is IMethodSymbol invokedMethod &&
+                    invokedMethod.ContainingType.SpecialType == SpecialType.System_String &&
+                    TargetMethods.Contains(invokedMethod.Name) &&
+                    invokedMethod.Parameters.Length > 0 &&
+                    invokedMethod.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                {
+                    method = invokedMethod.Name;
+
+                    foreach (var argument in invocationOperation.Arguments)
+                    {
+                        if (argument.Value.Type != null &&
+                            argument.Value.Type.Equals(_stringComparisonType) &&
+                            argument.Value is IFieldReferenceOperation fieldReferenceOperation)
+                        {
+                            if (fieldReferenceOperation.Field.Equals(_ordinalStringComparisonSymbol))
+                            {
+                                comparison = StringComparisonUsed.Ordinal;
+                            }
+                            else if (fieldReferenceOperation.Field.Equals(_invariantCultureStringComparisonSymbol))
+                            {
+                                comparison = StringComparisonUsed.InvariantCulture;
+                            }
+                            else
+                            {
+                                comparison = StringComparisonUsed.Other;
+                            }
```

Comment:
StartsWith and EndWidth have overloads that takes cultureInfo parameter. We need to mark these as `StringComparisonUsed.Other`

Reply:
What if `CultureInfo.InvariantCulture` is being used? Example:
```cs
"test".StartsWith("a", ignoreCase: false, culture: CultureInfo.InvariantCulture);
```
Should it be marked as `StringComparisonUsed.InvariantCulture` in this case?